src/main/drivers/dshot.c

```diff
@@ -151,6 +151,27 @@ FAST_DATA_ZERO_INIT static float erpmToHz;
 FAST_DATA_ZERO_INIT static float dshotRpmAverage;
 FAST_DATA_ZERO_INIT static float dshotRpm[MAX_SUPPORTED_MOTORS];
 
+// Lookup table for extended telemetry type decoding
+// Only contains extended telemetry types, eRPM is handled by conditional logic
+static const dshotTelemetryType_t extendedTelemetryLookup[9] = {
+    DSHOT_TELEMETRY_TYPE_eRPM,
+    // Temperature range (in degree Celsius, just like Blheli_32 and KISS)
+    DSHOT_TELEMETRY_TYPE_TEMPERATURE,
+    // Voltage range (0-63,75V step 0,25V)
+    DSHOT_TELEMETRY_TYPE_VOLTAGE,
+    // Current range (0-255A step 1A)
+    DSHOT_TELEMETRY_TYPE_CURRENT,
+    // Debug 1 value
+    DSHOT_TELEMETRY_TYPE_DEBUG1,
+    // Debug 2 value
+    DSHOT_TELEMETRY_TYPE_DEBUG2,
+    // Debug 3 value
+    DSHOT_TELEMETRY_TYPE_DEBUG3,
+    // State / events
+    DSHOT_TELEMETRY_TYPE_STATE_EVENTS,
+    DSHOT_TELEMETRY_TYPE_eRPM,  // Invalid/fallback
```

Comment:
This is not necessary (typeIndex < ARRAYLEN(extendedTelemetryLookup) handles that)
```suggestion
    DSHOT_TELEMETRY_TYPE_eRPM,  // Invalid/fallback
```